Cabal-syntax/src/Distribution/Fields/ParseResult.hs

```diff
@@ -17,15 +17,15 @@ module Distribution.Fields.ParseResult (
     withoutWarnings,
     ) where
 
-import           Distribution.Compat.Prelude
+import           Distribution.Compat.Prelude  hiding (Applicative(..))
```

Comment:
With this change, this module will not compile on base < 4.10.0. Not sure what to do here.

Reply:
(other than introduce more CPP)

Reply:
I think CPP is the only resort.
(BTW it would be nice to mention such situation in the migration guide)

Reply:
I added it to the migration guide, but I'm not sure I (if it's possible at all) to provide instructions that cover most of the cases, it feels like this might be very varying in what the fix is.

Reply:
I tried to push a fix for this but it's not trivial for me to compile with base <4.10 right now, could you take a look as well?